### PR TITLE
BZ-1278596 - Signal subprocess is never invoked

### PR DESCRIPTION
**Pull request by jesuino**

### CODE REVIEW

**Comment by mswiderski** [comment]
@jesuino we need to add test to make sure that the changes does solve the problem for both types of signal event handling - with signal element and without. do you mind adding some tests for these cases?

**Comment by jesuino** [comment]
Hello Maciej, sorry, forgot to send it, will do in a few hours
Em 12/11/2015 03:51, "Maciej Swiderski" <notifications@github.com> escreveu:

**Comment by jesuino** [comment]
Hello @mswiderski I just added the test. It asserts if a node inside the signal event subprocess was triggered. Let me know if it is correct. Thanks!

**Comment by mswiderski** [comment]
Looks good, thanks @jesuino

**Comment by mswiderski** [comment]
ok to test

File: jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/BoundaryEventHandler.java [hunk 1]
```diff
@@ -338,6 +339,16 @@ protected void handleSignalNode(final Node node, final Element element,
             } else if ("signalEventDefinition".equals(nodeName)) {
                 String type = ((Element) xmlNode).getAttribute("signalRef");
                 if (type != null && type.trim().length() > 0) {
+                    Map<String, Signal> signals = (Map<String, Signal>) ((ProcessBuildData) parser
+                            .getData()).getMetaData("Signals");
+                    
+                    if (signals != null && signals.containsKey(type)) {
+                        Signal signal = signals.get(type);                      
+                        type = signal.getName();
+                        if (type == null) {
```
Comment (mrietveld) [inline jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/BoundaryEventHandler.java]: This should be in the RuleFlowProcessValidator class (checking for type).

File: jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/StartEventHandler.java [hunk 2]
```diff
@@ -102,6 +103,15 @@ protected void handleNode(final Node node, final Element element, final String u
                 break;
             } else if ("signalEventDefinition".equals(nodeName)) {
                 String type = ((Element) xmlNode).getAttribute("signalRef");
+                Map<String, Signal> signals = (Map<String, Signal>) ((ProcessBuildData) parser.getData()).getMetaData("Signals");
+                
+                if (signals != null && signals.containsKey(type)) {
+                    Signal signal = signals.get(type);                      
+                    type = signal.getName();
+                    if (type == null) {
+                        throw new IllegalArgumentException("Signal definition must have a name attribute");
```
Comment (mrietveld) [inline jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/StartEventHandler.java]: Please move to the RuleFlowProcessValidator class, thanks!

Comment (jesuino) [inline jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/StartEventHandler.java]: Hello Marco,

I took this code change from these commits [1][2] from branch 6.3.x and the
change was in the IntermediateThrowEventHandler class. Am I missing
something?



[1]
https://github.com/droolsjbpm/jbpm/commit/1098074db5d0e39830267669970b9160e903c332
[2]
https://github.com/droolsjbpm/jbpm/commit/cadeff2c15305c4c6ce2b0c539f876cbb6a1913a

--
*William Antônio Siqueira*
*Java Support Analyst*

*http://fxapps.blogspot.com <http://fxapps.blogspot.com>*
*http://www.williamantonio.wordpress.com
<http://www.williamantonio.wordpress.com>*
*http://williamprogrammer.com <http://williamprogrammer.com>*

2015-11-23 8:35 GMT-02:00 Marco Rietveld <notifications@github.com>:

Comment (mrietveld) [inline jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/StartEventHandler.java]: Argh, okay. Leave them here. I'll move them to the RuleFlowProcessValidator class later.

**Comment by mswiderski** [comment]
merged into 6.2.x (https://github.com/droolsjbpm/jbpm/commit/00e930fa4f3cf3a9d71fbb6b7dd8876a63a85a93) with some missing bits and squashed commits into one, thanks @jesuino

**Pull request closed without merging**